chitchat/src/types.rs

```diff
@@ -108,7 +133,7 @@ impl PartialEq for VersionedValue {
     fn eq(&self, other: &Self) -> bool {
         self.value.eq(&other.value)
             && self.version.eq(&other.version)
-            && self.is_tombstone().eq(&other.is_tombstone())
+            && self.is_deleted().eq(&other.is_deleted())
```

Comment:
I'm not sure is_deleted is enough to qualify equality. `DeletionStatus::Set` and `DeletionStatus::DeleteAfterTtl` are considered equal by this statement. If that is intended, a comment should explain why

Reply:
Thx. I switched to checking the discriminant, added a comment, and made PartialEq cfg test only.